problem-domain.html

```diff
@@ -0,0 +1,122 @@
+<!DOCTYPE>
+<html>
+  <canvas id="myCanvas" width=1000 height=700> </canvas>
+</hmtl>
+
+<script>
+// several lines of code relating to using canvas and the context variable were adapted from
+// http://www.w3schools.com/html/html5_canvas.asp
+
+  function Point(x, y) {
+    this.x = x;
+    this.y = y;
+  };
+
+  function Box(origin, width, height) {
+    this.origin = origin;
+    this.height = height;
+    this.width = width;
+  };
+
+  function Arrow(startPoint, endPoint) {
+    this.startPoint = startPoint;
+    this.endPoint = endPoint;
+  };
+
+  function UMBDiagram() {
+    var el = document.getElementById("myCanvas");
+    this.context = el.getContext("2d");
+    this.boxes = [];
+    this.arrows = [];
+  }
+
+  drawLine = function(context, startPoint, endPoint) {
```

Comment:
oh, i see. lot's of objects use the line...so, a Line object that the other classes include and use.